### PR TITLE
[DependencyInjection] Document the Required attribute

### LINKED ISSUE #14189: [DependencyInjection] Add the Required attribute
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#37545
| PR author(s) | @derrabus
| Merged in    | 5.2-dev

### PR DESCRIPTION
**Pull request by derrabus**
This PR adds documentation for symfony/symfony#37545 and fixes #14189.

### CODE REVIEW

File: service_container/autowiring.rst [hunk 1]
```diff
@@ -546,48 +546,105 @@ When autowiring is enabled for a service, you can *also* configure the container
 to call methods on your class when it's instantiated. For example, suppose you want
 to inject the ``logger`` service, and decide to use setter-injection::
```
Comment (wouterj) [inline service_container/autowiring.rst]:
```suggestion
to inject the ``logger`` service, and decide to use setter-injection:
```

(double colon indicates the next block is a PHP code block, but that's no longer true - it's a configuration block now)

File: service_container/autowiring.rst [hunk 2]
```diff
@@ -546,48 +546,105 @@ When autowiring is enabled for a service, you can *also* configure the container
...
+the ``@required`` annotation instead.
 
-    namespace App\Util;
+.. versionadded:: 5.2
 
-    class Rot13Transformer
-    {
-        /** @required */
-        public LoggerInterface $logger;
+    The `#[Required]` attribute was introduced in Symfony 5.2.
```
Comment (wouterj) [inline service_container/autowiring.rst]:
```suggestion
    The ``#[Required]`` attribute was introduced in Symfony 5.2.
```

**Review by wouterj — APPROVED** [review]
Thanks! Two minor syntax things that we can also fix while merging.

File: service_container/autowiring.rst [hunk 3]
```diff
@@ -544,50 +544,107 @@ Autowiring other Methods (e.g. Setters and Public Typed Properties)
...
+
+    .. code-block:: php-attributes
+
+        namespace App\Util;
+
+        use Symfony\Contracts\Service\Attribute\Required;
+
+        class Rot13Transformer
+        {
+            [#Required]
```
Comment (derrabus) [inline service_container/autowiring.rst]:
```suggestion
            #[Required]
```

File: service_container/autowiring.rst [hunk 4]
```diff
@@ -544,50 +544,107 @@ Autowiring other Methods (e.g. Setters and Public Typed Properties)
...
+
+        use Symfony\Contracts\Service\Attribute\Required;
+
+        class Rot13Transformer
         {
-            $this->logger->info('Transforming '.$value);
-            // ...
+            private $logger;
+
+            [#Required]
```
Comment (derrabus) [inline service_container/autowiring.rst]:
```suggestion
            #[Required]
```

**Comment by derrabus** [comment]
Rebased. Let's see if @Nyholm's changes fix the build. 😃

**Comment by derrabus** [comment]
The build is green. 🎉

**Comment by Nyholm** [comment]
I can’t believe you doubted me.

:)

**Comment by javiereguiluz** [comment]
Thanks Alexander! I'm so excited about PHP 8 attributes, that I love all the contributions you are making to introduce them in Symfony. Thanks a lot!

**Comment by derrabus** [comment]
Thank you, @javiereguiluz! If you want to see the attributes in action, have a look at symfony/demo#1169. 😃

**Pull request merged**